maxAreaOfIsland.js

```diff
@@ -0,0 +1,126 @@
+/*
+You are given an m x n binary matrix grid. An island is a group of 1's (representing land) connected 4-directionally (horizontal or vertical.) 
+You may assume all four edges of the grid are surrounded by water.
+
+The area of an island is the number of cells with a value 1 in the island.
+
+Return the maximum area of an island in grid. If there is no island, return 0.
+*/
+
+//    0  1  2  3  4  5  6  7  8  9  10 11 12
+// 0 [0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
+// 1 [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
+// 2 [0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+// 3 [0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 0],
+// 4 [0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0],
+// 5 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+// 6 [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0],
+// 7 [0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0],
+
+const { Queue } = require('./utils/queue');
+
+const directions = [
+    [-1, 0],
+    [0, -1],
+    [0, 1],
+    [1, 0],
+];
+
+const ISLAND = 1;
+const WATER = 0;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isWithinBounds = (row, col, grid) => {
+    const numRow = grid.length;
+    const numCol = grid[0].length;
+
+    const isRowInBound = row >= 0 && row < numRow;
+    const isColInBound = col >= 0 && col < numCol;
+
+    return isRowInBound && isColInBound;
+};
+
+const getNeighbors = (row, col, grid) => {
+    const potentialNeighbors = [];
+
+    for (const direction of directions) {
+        const [rowChange, colChange] = direction;
+
+        const newRow = row + rowChange;
+        const newCol = col + colChange;
+
+        if (!isWithinBounds(newRow, newCol, grid)) continue;
+
+        if (grid[newRow][newCol] === WATER) continue;
+
+        potentialNeighbors.push([newRow, newCol]);
+    }
+    return potentialNeighbors;
+};
+
+const getIslandArea = (row, col, grid, visited) => {
+    const queue = new Queue();
+    const rowColString = getRowColString(row, col);
+
+    visited.add(rowColString);
+    queue.enqueue([row, col]);
+
+    let getIslandArea = 0;
```

Comment:
`getIslandArea` sounds like it should be a function (actually, it is the name of your function). You should name this something else.